main.go

```diff
@@ -301,19 +319,30 @@ func watchResource(c *cli.Context, r *ObjectResource) error {
 
 	og := r.DeploymentStatus.ObservedGeneration
 	ready := false
-	var availableResourceCount int32 = 0
-	var unavailableResourceCount int32 = 0
+	var availableResourceCount int32
+	var unavailableResourceCount int32
 
 	for {
 		select {
 		case <-timeout:
 			return fmt.Errorf("%s rolling update %q timed out after %s", r.Kind, r.Name, c.Duration("timeout").String())
 		case <-ticker.C:
 			r.DeploymentStatus = DeploymentStatus{}
-			// @TODO should a one-off error (perhaps network issue) cause us to completly fail?
-			if err := updateResourceStatus(c, r); err != nil {
-				return err
+
+			// Retry on error until max retries is met
+			for attempt := 0; attempt < MaxHealthcheckRetries; attempt++ {
```

Comment:
rather than doing this in quick succession on a failure, perhaps 
```go
if err := updateResourceStatus(c, r); err != nil {
   if someFailedCounter >= MaxHealthcheckRetries {
     return err
   }
   someFailedCounter += 1
   continue  // and wait for the next check-interval
}
someFailedCounter = 0

```
So by default it will give three attempts 1 second apart .. what are your thoughts?

Reply:
beyond the above .. LGTM

Reply:
Could add a time.Sleep in the loop if you'd like?